addons/crm/models/crm_lead.py

```diff
@@ -568,8 +568,11 @@ def _prepare_contact_name_from_partner(self, partner):
 
     def _prepare_partner_name_from_partner(self, partner):
         partner_name = partner.parent_id.name
-        if not partner_name and partner.is_company:
-            partner_name = partner.name
+        if not partner_name:
```

Comment:
I would avoid indenting code and keep a single level of if 
```
if not partner_name and partner.is_company:
    # blah
elif not partner_name and partner.company_name:
    # blah
```

Moreover this change should be in its own commit and be covered by unit tests. We already have quite a lot of tests for partner / lead synchronization so this should be quite easy to write.